src/openms/source/CHEMISTRY/SpectrumAnnotator.cpp

```diff
@@ -0,0 +1,425 @@
+          {
+            fragmenterrors.push_back(error_annotations.at(i));
+            intensities.push_back(spec[i].getIntensity());
+            match_intensity += spec[i].getIntensity();
+            mzs.push_back(spec[i].getMZ());
+            const String& ion_name = type_annotations.at(i);
+            {
+              ions.push_back(ion_name);
+              if (terminal_series_match_ratio_)
+              {
+                if (boost::regex_match(ion_name, nt_regex_))
+                {
+                  nint += spec[i].getIntensity();
+                }
+                else if (boost::regex_match(ion_name, ct_regex_))
+                {
+                  cint += spec[i].getIntensity();
+                }
+              }
+              if (max_series_)  // without loss max series is sometimes pretty crummy
+              {
+                const String& ion_type = ion_name.prefix(1);
+                boost::cmatch what;
+                if (boost::regex_match(ion_name.c_str(), what, seriesposition_regex_) &&
+                        ListUtils::contains(allowed_types, ion_type))
+                {
+                  // what[0] contains the whole string
+                  // what[1] contains the response code
+                  try
+                  {
+                    int i = std::atoi(what[1].first);
+                    ion_series[ion_type].at(i-1) = true;
+                  }
+                  catch (std::out_of_range)
+                  {
+                    LOG_WARN << "Note: Ions of " << ion_type << ion_name.substr(1).remove('+').toInt()
+                             << " will be ignored for max_series " << ph->getSequence().toString() << endl;
+                    continue;
+                  }
+                }
+              }
+            }
+          }
+        }
+        if (basic_statistics_)
+        {
+          ph->setMetaValue("matched_ions", ListUtils::concatenate(ions, ","));
+          ph->setMetaValue("matched_intensity", match_intensity);
+          ph->setMetaValue("matched_ion_number", ions.size());
+          ph->setMetaValue("peak_number", spec.size());
+          ph->setMetaValue("sum_intensity", sum_intensity);
+        }
+        if (terminal_series_match_ratio_)
+        {
+          ph->setMetaValue("NTermIonCurrentRatio", nint/match_intensity);
+          ph->setMetaValue("CTermIonCurrentRatio", cint/match_intensity);
+        }
+        if (topNmatch_fragmenterrors_)
+        {
+          if (fragmenterrors.empty())
+          {
+            ph->setMetaValue("median_fragment_error", 0);
+            ph->setMetaValue("IQR_fragment_error", 0);
+            ph->setMetaValue("topN_meanfragmenterror", 0);
+            ph->setMetaValue("topN_MSEfragmenterror", 0);
+            ph->setMetaValue("topN_stddevfragmenterror", 0);
+          }
+          else
+          {
+            vector<double> fe(fragmenterrors);
+            std::size_t mid = fe.size()/2;
```

Comment:
spaces around operators here and several places.
we usually use Size instead of size_t in OpenMS